File: ts/samples/dapr-only/src/counter-actor.ts
import { AbstractActor } from '@dapr/dapr';

/**
 * Simple counter actor that maintains a count in state
 */
export class CounterActor extends AbstractActor {
  private readonly STATE_KEY = 'count';

  async increment(): Promise<number> {
    console.log(`[CounterActor ${this.id}] increment() called`);
    
    const stateManager = await this.getStateManager();
    const [hasState, currentCount] = await stateManager.tryGetState<number>(this.STATE_KEY);
    
    const count = hasState ? currentCount! : 0;
    const newCount = count + 1;
    
    await stateManager.setState(this.STATE_KEY, newCount);
    console.log(`[CounterActor ${this.id}] Count incremented from ${count} to ${newCount}`);
    
    return newCount;
  }

  async decrement(): Promise<number> {
    console.log(`[CounterActor ${this.id}] decrement() called`);
    
    const stateManager = await this.getStateManager();
    const [hasState, currentCount] = await stateManager.tryGetState<number>(this.STATE_KEY);
    
    const count = hasState ? currentCount! : 0;
    const newCount = count - 1;
    
    await stateManager.setState(this.STATE_KEY, newCount);
    console.log(`[CounterActor ${this.id}] Count decremented from ${count} to ${newCount}`);
    
    return newCount;
  }

  async getCount(): Promise<number> {
    console.log(`[CounterActor ${this.id}] getCount() called`);
    
    const stateManager = await this.getStateManager();
    const [hasState, currentCount] = await stateManager.tryGetState<number>(this.STATE_KEY);
    
    const count = hasState ? currentCount! : 0;
    console.log(`[CounterActor ${this.id}] Current count is ${count}`);
    
    return count;
  }

  async reset(): Promise<void> {
    console.log(`[CounterActor ${this.id}] reset() called`);
    
    const stateManager = await this.getStateManager();
    await stateManager.setState(this.STATE_KEY, 0);
    
    console.log(`[CounterActor ${this.id}] Count reset to 0`);
  }

  async onActivate(): Promise<void> {
    console.log(`[CounterActor ${this.id}] Activated`);
  }

  async onDeactivate(): Promise<void> {
    console.log(`[CounterActor ${this.id}] Deactivated`);
  }
}

File: pure/internalUsages/DaprSample2/CounterActor.cs
using Dapr.Actors.Runtime;

namespace DaprSample2;

/// <summary>
/// Counter Actor implementation
/// </summary>
public class CounterActor : Actor, ICounterActor
{
    private const string CounterStateName = "counter";

    public CounterActor(ActorHost host) : base(host)
    {
    }

    /// <summary>
    /// Get counter value
    /// </summary>
    public async Task<int> GetCountAsync()
    {
        Logger.LogInformation("Counter Actor {ActorId} GetCountAsync called", Id);
        var count = await StateManager.TryGetStateAsync<int>(CounterStateName);
        var result = count.HasValue ? count.Value : 0;
        Logger.LogInformation("Counter Actor {ActorId} returning count: {Count}", Id, result);
        return result;
    }

    /// <summary>
    /// Increment counter
    /// </summary>
    public async Task IncrementAsync()
    {
        var currentCount = await GetCountAsync();
        var newCount = currentCount + 1;
        await StateManager.SetStateAsync(CounterStateName, newCount);
        
        Logger.LogInformation("Counter {ActorId} incremented to {Count}", Id, newCount);
    }

    /// <summary>
    /// Reset counter
    /// </summary>
    public async Task ResetAsync()
    {
        await StateManager.SetStateAsync(CounterStateName, 0);
        Logger.LogInformation("Counter {ActorId} reset to 0", Id);
    }

    /// <summary>
    /// Processing when Actor is activated
    /// </summary>
    protected override Task OnActivateAsync()
    {
        Logger.LogInformation("Counter Actor {ActorId} activated", Id);
        return Task.CompletedTask;
    }

    /// <summary>
    /// Processing when Actor is deactivated
    /// </summary>
    protected override Task OnDeactivateAsync()
    {
        Logger.LogInformation("Counter Actor {ActorId} deactivated", Id);
        return Task.CompletedTask;
    }
}
